onnxruntime/core/mlas/lib/qgemm.cpp

```diff
@@ -210,9 +210,9 @@ MlasDynamicQGemmBatch (
     MLAS_THREADPOOL* ThreadPool
 ) {
 #if defined(USE_KLEIDIAI) && !defined(_MSC_VER)
-    //No fallback and putting in guards
-    if(MLAS_CPUIDINFO::GetCPUIDInfo().HasArm_SME()){
-    ArmKleidiAI::MlasDynamicQGemmBatch(Shape, DataParams, BatchN, ThreadPool);
+    //No fallback and putting in guards. This implementation is SME2 specific.
+    if(ArmKleidiAI::UseSME2){
```

Comment:
Does this need a corresponding chhange here too - https://github.com/microsoft/onnxruntime/blob/c30905d638418383b8d83b3b1bb65b7b42226f5a/onnxruntime/contrib_ops/cpu/quantization/dynamic_quantize_matmul.cc#L218 ?

Seeing as there is no MLAS fallback, we would like to gate things appropriately in the upper CPU EP kernel.